src/main/java/uk/ac/cam/cl/dtg/teaching/pottery/controllers/RepoController.java

```diff
@@ -94,24 +89,55 @@ public RepoInfo makeRemoteRepo(
       if (!c.getInfo().getVariants().contains(variant)) {
         throw new TaskMissingVariantException("Variant " + variant + " is not defined");
       }
-      Repo r = repoFactory.createInstance(taskId, usingTestingVersion, expiryDate, variant, remote);
-      RepoInfo info = r.toRepoInfo();
-      if (!info.isRemote()) {
-        r.copyFiles(c);
-      }
-      return info;
     }
+    Repo r = repoFactory.createInstance(taskId, usingTestingVersion, null, variant, remote);
+    final String repoId = r.getRepoId();
+    worker.schedule(
+        new Job() {
+          @Override
+          public int execute(
+              TaskIndex taskIndex,
+              RepoFactory repoFactory,
+              ContainerManager containerManager,
+              Database database) {
+            try {
+              Task t = taskIndex.getTask(taskId);
+              try (TaskCopy c =
+                  usingTestingVersion ? t.acquireTestingCopy() : t.acquireRegisteredCopy()) {
+                repoFactory.initialiseInstance(repoId, c, validityMinutes);
+              }
+            } catch (TaskNotFoundException | RepoNotFoundException | RepoExpiredException
+                | RepoStorageException e) {
+              LOG.error("Failed to initialise repository", e);
```

Comment:
it would be nice to able to surface this message through the api to the caller

Reply:
Do you mean add a FAILED state to the repo and a nullable error message that could store the exception message and return it? Or something else? In theory, none of these exceptions should occur, and some of them prevent writing to the repo anyway.

Reply:
Nullable error message would be good I think. e.g. when you try to update registered version it will pass back to you what goes wrong when you poll registration status.

Reply:
K, will add that.